ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -182,6 +158,62 @@ S3FileSystemTerminate(WT_FILE_SYSTEM *fileSystem, WT_SESSION *session)
     return (0);
 }
 
+/*
+ * S3DirectoryList --
+ *     Return a list of object names for the given location.
+ */
+static int
+S3DirectoryList(WT_FILE_SYSTEM *fileSystem, WT_SESSION *session, const char *directory,
+  const char *prefix, char ***dirlistp, uint32_t *countp)
+{
+    S3_FILE_SYSTEM *fs;
+    fs = (S3_FILE_SYSTEM *)fileSystem;
+    std::vector<std::string> objects;
+    int ret = 0;
+    if (ret =
+          fs->conn->ListObjects(std::string(directory), std::string(prefix), objects, *countp) != 0)
+        goto err;
+    std::cout << "Objects in bucket '" << directory << "':" << std::endl;
+    if (!objects.empty()) {
```

Comment:
Here, can we do the following instead:
```suggestion
   for (const auto &object : objects)
     std::cout << "  * " << object << std::endl;
   if(objects.empty())
     std::cout << "No objects in bucket." << std::endl;
```